projects/storefrontlib/src/cms-components/checkout/components/shipping-address/shipping-address.component.ts

```diff
@@ -23,77 +27,131 @@ export interface CardWithAddress {
+  }
+
+  get selectedAddress$(): Observable<Address> {
+    return this.checkoutDeliveryService.getDeliveryAddress().pipe(
+      tap((address) => {
+        if (
+          address &&
+          (this.selectedAddress === undefined ||
+            this.selectedAddress.id !== address.id)
+        ) {
+          this.selectedAddress = address;
+          if (this.forceLoader) {
+            this.next();
+          }
+        }
+      })
+    );
+  }
 
-    this.cards$ = combineLatest([
-      this.existingAddresses$,
+  get cards$(): Observable<CardWithAddress[]> {
+    return combineLatest([
+      this.getSupportedAddresses(),
       this.selectedAddress$,
       this.translation.translate('checkoutAddress.defaultShippingAddress'),
       this.translation.translate('checkoutAddress.shipToThisAddress'),
       this.translation.translate('addressCard.selected'),
     ]).pipe(
-      map(
-        ([
-          addresses,
-          selected,
-          textDefaultShippingAddress,
-          textShipToThisAddress,
-          textSelected,
-        ]) => {
-          // Select default address if none selected
-          if (
-            addresses.length &&
-            (!selected || Object.keys(selected).length === 0)
-          ) {
-            const defaultAddress = addresses.find(
-              (address) => address.defaultAddress
-            );
-            selected = defaultAddress;
-            this.selectAddress(defaultAddress);
-          }
-          return addresses.map((address) => {
-            const card = this.getCardContent(
-              address,
-              selected,
-              textDefaultShippingAddress,
-              textShipToThisAddress,
-              textSelected
-            );
-            return {
-              address,
-              card,
-            };
-          });
-        }
+      tap(([addresses, selected]) =>
+        this.selectDefaultAddress(this.isAccount, addresses, selected)
+      ),
+      map(([addresses, selected, textDefault, textShipTo, textSelected]) =>
+        (<any>addresses).map((address) => ({
+          address,
+          card: this.getCardContent(
+            address,
+            selected,
+            textDefault,
+            textShipTo,
+            textSelected
+          ),
+        }))
       )
     );
+  }
 
-    if (!this.activeCartService.isGuestCart()) {
-      this.userAddressService.loadAddresses();
+  getSupportedAddresses(): Observable<Address[] | B2BAddress[]> {
+    if (this.isAccount) {
+      let costCenterId: string;
+      this.checkoutCostCenterService
+        .getCostCenter()
+        .pipe(take(1))
+        .subscribe((selected) => (costCenterId = selected));
+      return this.userCostCenterService.getCostCenterAddresses(costCenterId);
     } else {
-      this.isGuestCheckout = true;
+      return this.userAddressService.getAddresses();
+    }
+  }
+
+  selectDefaultAddress(
+    isAccount: boolean,
```

Comment:
Suggestion: For consistency, perhaps we standardize one way or the other the functions in the component: `getSupportedAddresses()` uses `this.isAccount` directly, while selectDefaultAddress takes a  `isAccount: boolean` parameter.